main.tf

```diff
@@ -8,8 +8,12 @@ resource "aws_s3_bucket" "default" {
   policy        = var.policy
   tags          = module.this.tags
 
-  versioning {
-    enabled = var.versioning_enabled
+  dynamic "versioning" {
+    for_each = var.bucket_versioning ? [1] : [0]
```

Comment:
Why have two separate boolean variables here?

Reply:
Long time ago, but as I remember...  cause versioning can be enabled or disabled for bucket. In case versioning is enabled, it could be set to [suspended](https://docs.aws.amazon.com/AmazonS3/latest/userguide/VersionSuspendedBehavior.html).

Reply:
I don't think that's relevant here tho as you can still suspend versioning on the bucket by toggling `version_enabled = false` if `versioning_enabled` was used in the `for_each` instead of `bucket_versioning`.

Reply:
Hmm after reading the issue you linked to, perhaps I'm mistaken. This seems to be a workaround for an upstream bug but I don't know if we want to support this workaround or wait for the bug to be fixed upstream.